### PR TITLE
Ranking Implemented

### PR DESCRIPTION
**Pull request by pablojor**

### CODE REVIEW

File: src/BulletHurtPlayerEC.cpp [hunk 1]
```diff
@@ -26,9 +27,14 @@ void BulletHurtPlayerEC::checkEvent() {
         LifeC* playerHealth = dynamic_cast<LifeC*>(
             scene->getEntitybyId("Player")->getComponent("LifeC"));
 
-        // if player dies sleep method is called
-        if (playerHealth->doDamage(damage))
-            ;
+        // if player dies sleep method is called and Ranking updates
+        if (playerHealth->doDamage(damage)) {
+
+            dynamic_cast<RankingManagerC*>(
```
Comment (kyranet) [inline src/BulletHurtPlayerEC.cpp]:
```suggestion
            reinterpret_cast<RankingManagerC*>(
```

File: src/MeleeEnemyBehaviourEC.cpp [hunk 2]
```diff
@@ -27,8 +28,13 @@ void MeleeEnemyBehaviourEC::checkEvent() {
                 LifeC* playerHealth = dynamic_cast<LifeC*>(
                     scene->getEntitybyId("Player")->getComponent("LifeC"));
                 // if player dies sleep method is called
-                if (playerHealth->doDamage(getAttack()))
-                    ;
+                if (playerHealth->doDamage(getAttack())) {
+
+                    dynamic_cast<RankingManagerC*>(
```
Comment (kyranet) [inline src/MeleeEnemyBehaviourEC.cpp]:
```suggestion
                    reinterpret_cast<RankingManagerC*>(
```

File: bin/Files/ranking.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+1 dsfg 10
+2 sdf 9
+3 sdf 8
+4 dfsg 7
+5 ddfhg 6
+6 gh 5
+7 fgh 4
+8 fgh 3
+9 fdg 2
+10 sdf 1
```
Comment (kyranet) [inline bin/Files/ranking.txt]: Untrack this file from remote and add it to `.gitignore`.

File: src/RankingManagerC.cpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    myfile.close();
+}
+
+void RankingManagerC::playerDied() {
+
+    RankingPosition rank;
+
+    rank.name = "name";
+    rank.round =
+        dynamic_cast<RoundManagerEC*>(father->getComponent("RoundManagerEC"))
```
Comment (kyranet) [inline src/RankingManagerC.cpp]: *Might need formatting:
```suggestion
        reinterpret_cast<RoundManagerEC*>(father->getComponent("RoundManagerEC"))
```

**Review by kyranet — DISMISSED** [review]

File: src/RankingManagerC.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+class SpawnerEnemiesEC;
+class SpawnerEC;
+
+struct RankingPosition {
+    int round;
+    std::string name;
+
+    bool operator<(RankingPosition const& b) { return round > b.round; }
+};
```
Comment (kyranet) [inline src/RankingManagerC.h]: Just an idea, what if you define `friend`s for `std::ostream`? That way you'd be able to do:

```cpp
friend std::ostream& operator<<(std::ostream& out, RankingPosition& position) {
  out << position.round << ' ' << position.name << '\n';
  return out;
}
```

```cpp
std::cout << rankingPosition << '\n';
```

And that'd write the name (`-` if not set) and the points, likewise:

```cpp
std::cin >> rankingPosition;
```

Which would internally do:

```cpp
friend std::ostream& operator>>(std::ostream& in, RankingPosition& position) {
  in >> position.round >> position.name;
  return in;
}
```

More information: https://stackoverflow.com/questions/38093045/what-does-friend-stdostream-operatorstdostream-out-linkedlist-list

I think you can make `RankingManagerC` much tidier by leaving the serialization and deserialization process to the struct itself, following OOP's best practices about encapsulation.

Comment (pablojor) [reply src/RankingManagerC.h]: C2676	'<<' binario: 'std::ostream' no define este operador o una conversión a un tipo aceptable para el operador predefinido

**Review by MiriamLeis — APPROVED** [review]

**Review by GonzaPM7 — APPROVED** [review]

**Pull request merged**